### PR TITLE
Introduce "non-reentrant time" statistics view

### PR DESCRIPTION
**Pull request by rokopt**
This PR implements the enhancement proposed in #159 :  introducing to the Statistics window a radio button, "Non-reentrant time", which counts only the first appearance of any one zone on a thread's call stack.

Some notes:

- Per https://github.com/wolfpld/tracy/issues/159#issuecomment-767819118 , I haven't added this option to "Find Zone", only to "Statistics".
- Similarly, I haven't added this option to CSV export.
- For an explanation of the reason for and workings of the new `View::IsZoneReentry` methods, see https://github.com/wolfpld/tracy/issues/159#issuecomment-774457623 .

### CODE REVIEW

**Comment by wolfpld** [comment]
Please fix the merge conflicts.

**Comment by rokopt** [comment]
> Please fix the merge conflicts.

Fixed.

**Comment by wolfpld** [comment]
I don't like the current UI for time selection. Please change to combo box, e.g.:

obraz

**Comment by rokopt** [comment]
> Please change to combo box, e.g.:

I've pushed a change to do that; how does it look?

(It's verbose translating back and forth between the `int` that the combo box needs and the `enum class AccumulationMode`; I'm not sure whether you find it worth having the `enum class` anyway -- I could change that if you prefer.)

**Comment by wolfpld** [comment]
> (It's verbose translating back and forth between the int that the combo box needs and the enum class AccumulationMode; I'm not sure whether you find it worth having the enum class anyway -- I could change that if you prefer.)

You can replace these switches with a pair of explicit casts.

**Comment by wolfpld** [comment]
The "non-reentrant" option should not be available when displaying sampling statistics, as it is not implemented there.

> I've pushed a change to do that; how does it look?

Much better. I would additionally change:
- "Including children" to "With children".
- "Accumulation mode" to "Accumulation", "Timing", "Stats", or something like that (the shorter, the better).
- Add some additional spacing, so that the options are not clumped together:

```c++
ImGui::SameLine();
ImGui::Spacing();
ImGui::SameLine();
```

Probably would be good to add this also to the rest of checkboxes.

**Comment by rokopt** [comment]
> You can replace these switches with a pair of explicit casts.

Okay, done.

**Comment by rokopt** [comment]
> The "non-reentrant" option should not be available when displaying sampling statistics, as it is not implemented there.

Good catch -- I've suppressed it in that case.


> Much better. I would additionally change:
> 
> * "Including children" to "With children".
> * "Accumulation mode" to "Accumulation", "Timing", "Stats", or something like that (the shorter, the better).

Done.

> * Add some additional spacing, so that the options are not clumped together:
> 
> ```c++
> ImGui::SameLine();
> ImGui::Spacing();
> ImGui::SameLine();
> ```

Done.

> Probably would be good to add this also to the rest of checkboxes.

Okay, I've sprinkled spacing throughout some of the pre-existing options too.  I did that in a separate commit for easy reversion in case I didn't do it where you intended.  :)

File: server/TracyView.cpp [hunk 1]
```diff
@@ -12269,9 +12269,29 @@ struct SrcLocZonesSlim
...
+    if ( m_statMode != 0 && m_statAccumulationMode == AccumulationMode::NonReentrantChildren )
+    {
+        m_statAccumulationMode = AccumulationMode::SelfOnly;
+    }
+    int accumulationMode = static_cast<int>( m_statAccumulationMode );
+    ImGui::Combo( "##accumulationMode", &accumulationMode, accumulationModeTable );
+    m_statAccumulationMode = static_cast<AccumulationMode>( accumulationMode );
+    ImGui::SameLine();
+    ImGui::Spacing();
+    ImGui::SameLine();
```
Comment (wolfpld) [inline server/TracyView.cpp:12292]: I think it is invalid to force `SameLine` and `Spacing` here (and in the function prologue). It should be responsibility of the code calling this function to properly glue the item to its surroundings. Right now it seems that a `Separator` call was catched by `SameLine`:

obraz

Comment (rokopt) [reply server/TracyView.cpp]: > I think it is invalid to force `SameLine` and `Spacing` here (and in the function prologue). It should be responsibility of the code calling this function to properly glue the item to its surroundings. Right now it seems that a `Separator` call was catched by `SameLine`:

Good eye.  How does the latest look?

**Comment by rokopt** [comment]
Thank you for all your time and guidance and patience with this one!

**Pull request merged**